src/test/java/bio/terra/integration/AzureIntegrationTest.java

```diff
@@ -384,23 +378,47 @@ private void populateOmopTable() throws Exception {
   public void testSnapshotBuilder() throws Exception {
     populateOmopTable();
 
-    // Test getConcepts
-    var getConceptResponse = dataRepoFixtures.getConcepts(steward, datasetId, 2);
-    var concepts = getConceptResponse.getResult();
+    var concept1 =
+        new SnapshotBuilderConcept().name("concept1").id(1).count(22).code("11").hasChildren(false);
+    var concept3 =
+        new SnapshotBuilderConcept().name("concept3").id(3).count(24).code("13").hasChildren(true);
+
+    getConceptChildrenTest(concept1, concept3);
+    searchConceptTest(concept1);
+    getConceptHierarchyTest(concept1, concept3);
+  }
 
-    var concept1 = concepts.get(0);
-    var concept3 = concepts.get(1);
+  private void searchConceptTest(SnapshotBuilderConcept concept1) throws Exception {
+    var searchConceptsResult =
+        dataRepoFixtures.searchConcepts(steward, datasetId, 19, concept1.getName());
+    // A concept returned by search concepts always has hasChildren = true, even if it doesn't
+    // have children.
```

Comment:
Is this a bug or by design?

Reply:
It's by design. The idea here is that we want all queries to always return the same properties for a concept so the code that reads the query result doesn't need to check for missing fields, but since this property isn't used by the Search Concepts UI, we also want to avoid the cost of computing the real value for this query. So the code hardcodes this result as `true`:
https://github.com/DataBiosphere/jade-data-repo/blob/develop/src/main/java/bio/terra/service/snapshotbuilder/utils/SearchConceptsQueryBuilder.java#L80